test/snippet/core/detail/reflection.cpp

```diff
@@ -0,0 +1,23 @@
+#include <iostream>
+
+#include <seqan3/core/detail/reflection.hpp>
```

Comment:
Redundant how? I need the iostream for the `std::cout` call, and it isn't included in reflection.hpp (or any of its includes).

Reply:
Oh I guess you were talking about the second include. The code example has ` #include <seqan3/core/detail/reflection.hpp>` included in the example, so I wanted to add it as well. But it doesn't have `using namespace seqan3` in it, so I have to put that outside the snippet. For some reason when this is above the include, it doesn't seem to work correctly?

Reply:
if the `#includes` is within the namespace, all the code will be too, but since the file will have its own `namespace seqan3 {` declaration you end up with namespace `seqan3::seqan3::` :wink: 

The code example as it is now is confusing to the user I think because I would assume it to be complete but copy-and-pasting it would not work. So either the using namespace after the (one and only) include within the `[[]]` documentation block, or remove the outer `using namespace` and add `seqan3::` before the `detail::get_display_name` function.

Reply:
I can do that! But the larger question is: Are all of the **visible** sections of code snippet expected to be copy-and-pastable? Because there are some examples which rely on arbitrary structs and enums being defined, which aren't shown in the examples...